src/Nethermind/Nethermind.Consensus/Processing/ParallelProcessing/Locations.cs

```diff
@@ -0,0 +1,31 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Concurrent;
+using System.Threading;
+using Nethermind.Core;
+using Nethermind.Core.Collections;
+
+namespace Nethermind.Consensus.Processing.ParallelProcessing;
+
+public interface ILocations<in TKey>
+{
+    int GetLocation(TKey address);
+}
+
+public class Locations : ILocations<Address>, ILocations<StorageCell>
+{
+    private int _index = -1;
+    private readonly ConcurrentDictionary<AddressAsKey, int> _addressLocations = new();
```

Comment:
So this is the mapping that I mentioned elsewhere. I think, if we made it in a uniform way, we could make it potentially cheaper and totally alloc free (https://github.com/NethermindEth/nethermind/blob/7357f676685cf009c40dc84c8b46a8719a7ecba6/src/Nethermind/Nethermind.State/StorageValueMap.cs from #8482 )